src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -223,6 +240,203 @@ protected ActionResult execute() throws EntityDoesNotExistException {
+            String showCommentTo =
+                    getRequestParamValue(Const.ParamsNames.RESPONSE_COMMENTS_SHOWCOMMENTSTO + commentIndex);
+            String showGiverNameTo =
+                    getRequestParamValue(Const.ParamsNames.RESPONSE_COMMENTS_SHOWGIVERTO + commentIndex);
+            String commentId = commentsToUpdateId.get(commentIndex);
+            String updatedCommentText = commentsToUpdateText.get(commentIndex);
+            FeedbackResponseAttributes response =
+                    logic.getFeedbackResponse(questionIdsForComments.get(commentIndex),
+                            responseGiverMapForComments.get(commentIndex),
+                            responseRecipientMapForComments.get(commentIndex));
+            updateResponseComment(showCommentTo, showGiverNameTo, commentId, response, updatedCommentText);
+
+        }
+    }
+
+    private void updateResponseComment(String showCommentTo, String showGiverNameTo, String commentId,
+                                       FeedbackResponseAttributes response, String updatedCommentText) {
+        FeedbackResponseCommentAttributes feedbackResponseComment = FeedbackResponseCommentAttributes
+                .builder(courseId, feedbackSessionName, response.giver, new Text(updatedCommentText))
+                .withCreatedAt(Instant.now())
+                .withGiverSection(response.giverSection)
+                .withReceiverSection(response.recipientSection)
+                .build();
+
+        feedbackResponseComment.setId(Long.parseLong(commentId));
+
+        //Edit visibility settings
+        feedbackResponseComment.showCommentTo = new ArrayList<>();
+        if (showCommentTo != null && !showCommentTo.isEmpty()) {
+            String[] showCommentToArray = showCommentTo.split(",");
+            for (String viewer : showCommentToArray) {
+                feedbackResponseComment.showCommentTo.add(FeedbackParticipantType.valueOf(viewer.trim()));
+            }
+        }
+        feedbackResponseComment.showGiverNameTo = new ArrayList<>();
+        if (showGiverNameTo != null && !showGiverNameTo.isEmpty()) {
+            String[] showGiverNameToArray = showGiverNameTo.split(",");
+            for (String viewer : showGiverNameToArray) {
+                feedbackResponseComment.showGiverNameTo.add(FeedbackParticipantType.valueOf(viewer.trim()));
+            }
+        }
+
+        FeedbackResponseCommentAttributes updatedComment = null;
+        try {
+            updatedComment = logic.updateFeedbackResponseComment(feedbackResponseComment);
+            //TODO: move putDocument to task queue
+            logic.putDocument(updatedComment);
+        } catch (InvalidParametersException e) {
+            setStatusForException(e);
+            isError = true;
+        } catch (EntityDoesNotExistException e) {
+            isError = true;
+        }
+
+        if (!isError) {
+            statusToAdmin += "FeedbackSubmissionEditSaveAction:<br>"
```

Comment:
Put this into the "try" block and the `isError` can be omitted.

Reply:
Since value of isError can be changed by many functions in this file, we cannot be sure that isError is false by putting it in try block?

Reply:
I meant putting the block of codes into the "try" block, and you still keep the line `isError = true` when you catch errors, it won't affect the operation of the other parts that use the `isError` right?

Reply:
Got it. Done.